tests/testthat/test-write_config.R

```diff
@@ -104,3 +108,92 @@ test_that("write_config validates length of hub_path and config_path", {
     regexp = "Assertion on 'config_path' failed: Must have length 1."
   )
 })
+
+test_that("write_config autoboxing works", {
+  skip_if_offline()
+  temp_hub <- setup_test_hub()
+  setup_test_hub_with_config_dir(temp_hub)
+  config <- hubUtils::read_config_file(
+    config_path = test_path("testdata/tasks-append.json")
+  ) |> as_config()
+
+  # Move to temp hub working directory to use default hub_path "." setting.
+  original_wd <- getwd()
+  setwd(temp_hub)
+  # Check that autoboxing causes invalid schema to be written
+  write_config(config = config, silent = TRUE, autobox = FALSE)
+  expect_false(suppressMessages(validate_config()))
+
+  write_config(config = config, overwrite = TRUE, silent = TRUE)
+  expect_true(suppressMessages(validate_config()))
+  file_contents <- readLines(file.path(temp_hub, "hub-config/tasks.json"))
+  expect_snapshot(cat(file_contents, sep = "\n"))
+
+  # Re-reading and appending a new round to the config also works
+  config <- hubUtils::read_config(".") |> as_config()
+  config_plus_new_round <- append_round(config = config, create_new_round())
+
+  write_config(config_plus_new_round,
+    silent = TRUE,
+    autobox = FALSE, overwrite = TRUE
+  )
+  expect_false(suppressMessages(validate_config()))
+
+  write_config(config_plus_new_round, overwrite = TRUE, silent = TRUE)
+  expect_true(suppressMessages(validate_config()))
+  file_contents <- readLines(file.path(temp_hub, "hub-config/tasks.json"))
+  expect_snapshot(cat(file_contents, sep = "\n"))
+
+  setwd(original_wd)
+})
+
+test_that("write_config with autobox = FALSE does not box and issues warning", {
+  skip_if_offline()
+  rounds <- create_test_rounds()
+  config <- create_test_config(rounds)
+  temp_hub <- setup_test_hub()
+  setup_test_hub_with_config_dir(temp_hub)
+
+  # Move to temp hub working directory to use default hub_path "." setting.
+  original_wd <- getwd()
+  setwd(temp_hub)
+  expect_snapshot(
+    write_config(
+      config = config,
+      hub_path = temp_hub,
+      autobox = FALSE
+    )
+  )
+  result <- write_config(
+    config = config, silent = TRUE,
+    autobox = FALSE, overwrite = TRUE
+  )
+  expect_false(suppressMessages(validate_config()))
+  setwd(original_wd)
```

Comment:
Minor: using withr so that the working directory doesn't remain set if the non-testthat code fails

```suggestion
  withr::with_dir(temp_hub, {
    expect_snapshot(
      write_config(
        config = config,
        hub_path = temp_hub,
        autobox = FALSE
      )
    )
    result <- write_config(
      config = config, silent = TRUE,
      autobox = FALSE, overwrite = TRUE
    )
    expect_false(suppressMessages(validate_config()))
  })
```

Reply:
Nice! `withr` is definitely a package I should be much more proficient in but haven't work with as much. Will definitely make an effort to take a look and used it more going forward.

Reply:
I also went ahead and used `withr::with_tempdir()` in the `write_config()` examples